diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 23cb5cd8c9..b74335033f 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -502,7 +502,7 @@ endif()
 
 # flags for user binaries = flags for toolchain binaries + Lake
 set(LEANC_STATIC_LINKER_FLAGS " ${TOOLCHAIN_STATIC_LINKER_FLAGS} -lLake")
-if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+if(CMAKE_SYSTEM_NAME MATCHES "Linux|Android")
   set(LEANC_SHARED_LINKER_FLAGS " ${TOOLCHAIN_SHARED_LINKER_FLAGS} -Wl,--as-needed -lLake_shared -Wl,--no-as-needed")
 else()
   set(LEANC_SHARED_LINKER_FLAGS " ${TOOLCHAIN_SHARED_LINKER_FLAGS} -lLake_shared")
@@ -549,7 +549,7 @@ if(NOT CMAKE_SYSTEM_NAME MATCHES "Darwin")
   string(APPEND LEAN_EXTRA_LINKER_FLAGS " -lm")
 endif()
 
-if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+if(CMAKE_SYSTEM_NAME MATCHES "Linux|Android")
   if(BSYMBOLIC)
     string(APPEND LEANC_SHARED_LINKER_FLAGS " -Wl,-Bsymbolic")
     string(APPEND TOOLCHAIN_SHARED_LINKER_FLAGS " -Wl,-Bsymbolic")
@@ -583,7 +583,7 @@ elseif(CMAKE_SYSTEM_NAME MATCHES "Windows")
   )
 endif()
 
-if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+if(CMAKE_SYSTEM_NAME MATCHES "Linux|Android")
   string(APPEND LEAN_EXTRA_LINKER_FLAGS " -ldl")
 endif()
 
@@ -735,7 +735,7 @@ add_subdirectory(shell)
 
 # The relative path doesn't work once the CMakeLists.txt files have been copied into the stage0 dir.
 # Since stage0 needs no tests, we just ignore them instead of fixing the path.
-if(NOT STAGE EQUAL 0)
+if(NOT STAGE EQUAL 0 AND NOT CMAKE_SYSTEM_NAME MATCHES "Android")
   add_subdirectory(../tests "${CMAKE_CURRENT_BINARY_DIR}/tests")
 endif()
 
